Functions/xASL_io_ReadDataPar.m

```diff
@@ -155,30 +155,38 @@
 % Iterate over fields
 for iField=1:length(listFields)
 	
-	% Convert string to num
-	TempField = xASL_str2num(StructIn.(listFields{iField}));
-	
-	% Check if we got an imaginary number
-	if max(isnumeric(TempField))
-		if imag(TempField)~=0
-			TempField = StructIn.(listFields{iField}); % Do not convert complex numbers
+	% Checks if conversion should be attempted. This check is done also inside the xASL_str2num, however,
+	% it is good to do it also beforehand, because you avoid unnecessary processing of numeric strings which
+	% can result in false-warnings about numeric being given inside a string. 
+	% i.e. this check avoids extra processing and warnings eventhough functionally not necessary
+	if ~isempty(StructIn.(listFields{iField})) && ~isnumeric(StructIn.(listFields{iField}))
+		% Convert string to num
+		TempField = xASL_str2num(StructIn.(listFields{iField}));
+		
+		% Check if we got an imaginary number
```

Comment:
This check should probably go to `xASL_str2num`, in general we don't want text to convert to an imaginary number right?

Reply:
Right. But this is probably not really relevant for the other cases, so I slightly prefer to keep this here for this specific use.